cedar-policy-validator/src/types.rs

```diff
@@ -1273,27 +1385,82 @@ mod test {
     }
 
     #[test]
-    fn test_record_lub() {
+    fn test_record_undef_lub() {
         assert_least_upper_bound_empty_schema(
-            Type::any_record(),
-            Type::any_record(),
-            Some(Type::record_with_attributes(None)),
+            Type::record_with_attributes_open_attributes(None, true),
+            Type::primitive_string(),
+            None,
         );
 
         assert_least_upper_bound_empty_schema(
-            Type::any_record(),
-            Type::any_entity_reference(),
+            Type::record_with_attributes_open_attributes(None, false),
+            Type::primitive_string(),
             None,
         );
 
         assert_least_upper_bound_empty_schema(
-            Type::record_with_required_attributes([
-                ("foo".into(), Type::False),
-                ("bar".into(), Type::primitive_long()),
-            ]),
-            Type::any_entity_reference(),
+            Type::record_with_attributes(None),
+            Type::set(Type::primitive_boolean()),
             None,
         );
+    }
+
+    #[test]
+    fn test_record_lub() {
+        println!(
+            "{}",
+            Type::is_subtype(
+                &empty_schema(),
+                &Type::record_with_attributes_open_attributes(None, true),
+                &Type::record_with_attributes_open_attributes(None, false)
+            )
+        );
+
+        println!(
+            "{}",
+            Type::is_subtype(
+                &empty_schema(),
+                &Type::record_with_attributes_open_attributes(None, true),
+                &Type::record_with_attributes_open_attributes(None, true)
+            )
+        );
+        println!(
+            "{}",
+            Type::is_subtype(
+                &empty_schema(),
+                &Type::record_with_attributes_open_attributes(None, false),
+                &Type::record_with_attributes_open_attributes(None, true)
+            )
+        );
+        println!(
+            "{}",
+            Type::is_subtype(
+                &empty_schema(),
+                &Type::record_with_attributes_open_attributes(None, false),
+                &Type::record_with_attributes_open_attributes(None, false)
+            )
+        );
```

Comment:
Rather than printing here, you might want actual tests. Or at least add comments with what the expected output is.

Reply:
outdated